infrastructure/httpapi/handlers/crossfire.go

```diff
@@ -1,9 +1,162 @@
 package handlers
 
-/**
-Add crossfire HTTP Endpoint functions here
+import (
+	"fmt"
+	"strings"
 
-1. GetLatestRanks()
-2. GetParticipantByOperatorAddress()
+	"github.com/crypto-com/chain-indexing/appinterface/cosmosapp"
 
-**/
+	"github.com/valyala/fasthttp"
+
+	crossfire "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire"
+	crossfire_constants "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire/constants"
+	crossfire_validator_view "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire/view"
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	"github.com/crypto-com/chain-indexing/infrastructure/httpapi"
+	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+)
+
+type Crossfire struct {
+	logger applogger.Logger
+
+	validatorAddressPrefix string
+	consNodeAddressPrefix  string
+
+	cosmosAppClient              cosmosapp.Client
+	crossfireValidatorsView      *crossfire_validator_view.CrossfireValidators
+	crossfireValidatorsStatsView *crossfire_validator_view.CrossfireValidatorsStats
+	httpClient                   *crossfire.HTTPClient
+}
+
+func NewCrossfire(
+	logger applogger.Logger,
+	validatorAddressPrefix string,
+	consNodeAddressPrefix string,
+	cosmosAppClient cosmosapp.Client,
+	rdbHandle *rdb.Handle,
+	participantsListURL string,
+) *Crossfire {
+	return &Crossfire{
+		logger.WithFields(applogger.LogFields{
+			"module": "CrossfireHTTPHander",
+		}),
+
+		validatorAddressPrefix,
+		consNodeAddressPrefix,
+
+		cosmosAppClient,
+		crossfire_validator_view.NewCrossfireValidators(rdbHandle),
+		crossfire_validator_view.NewCrossfireValidatorsStats(rdbHandle),
+		crossfire.NewHTTPClient(participantsListURL),
+	}
+}
+
+// ListAllCrossfireValidators Fetch all validators and combine them
+func (handler *Crossfire) ListAllCrossfireValidators(ctx *fasthttp.RequestCtx) {
+	var err error
+
+	// Fetch Remote List
+	participantList, errFetching := handler.httpClient.Participants()
+	if errFetching != nil {
+		handler.logger.Errorf("error listing validators: %v", errFetching)
+		httpapi.InternalServerError(ctx)
+	}
+
+	// Fetch DB Crossfire Validators List
+	crossfireValidatorsDB, err := handler.crossfireValidatorsView.List()
+	if err != nil {
+		handler.logger.Errorf("error listing validators: %v", err)
+		httpapi.InternalServerError(ctx)
+		return
+	}
+
+	var finalList = []CrossfireValidatorDetails{}
+
+	for _, participant := range *participantList {
```

Comment:
A small suggestion, we can use a map to filter out the participated validators from  `crossfireValidatorsDB`, to avoid the O(N^2) nested loop

Reply:
If you consider going for a Map to filter, a quick reference could be [here](https://github.com/crypto-com/chain-indexing/blob/1ad76f851e1394547fe3375bd36e471abc7a5d63/appinterface/projection/crossfire/crossfire.go#L569-L590)

Reply:
Thanks will update accordingly

Reply:
@foreseaz 
I wonder the `getValidatorStats` count can never be 40000 times as there is a `IF(....)` check before it selects from DB. 
So there shall be exactly `200` calls ( Only for the matching ones )

However, Map is a fantastic approach, but wouldn't optimise here much. In case me understanding something wrong let me know

Reply:
Sure, the `if` check can ensure it as `200` calls. But it will be better to avoid the nested loop, in case future adding more logic in the loop I think.